### PR TITLE
Remote task scheduler complete

### PR DESCRIPTION
**Pull request by atlas5301**
this is the complete version ov multi_machine_scheduler(with all components added)
a full test is added in 90.multi_machine_schedule folder

### CODE REVIEW

File: python/freetensor/core/RPCTools/center.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+if os.path.exists('./machine_list'):
+    os.remove('./machine_list')
+
+
+def check_connection():
+    return True
+
+
+def register_machine(remoteInfo, sev_status):
+    """机器注册函数，从每个机器拉取网络地址和端口并且分配uuid，同时更新每个机器的本地列表"""
```
Comment (roastduck) [inline python/freetensor/core/RPCTools/center.py:19]: English please. The audience of FreeTensor is international academia.

File: python/freetensor/core/RPCTools/center.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+from email.header import make_header
+from sre_constants import SUCCESS
+from xmlrpc.server import SimpleXMLRPCServer
+import xmlrpc.client as client
+import sys, socket, uuid, os, time
+import threading
+
+List = {}
```
Comment (roastduck) [inline python/freetensor/core/RPCTools/center.py:8]: What is the `List`, and why did you name `dict` as `List` :)

Should be a more specific name, in `under_score` fassion.

**Comment by roastduck** [comment]
`RPCTool` can be renamed to `rpc_tools`.

File: python/freetensor/core/RPCTools/center.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+from email.header import make_header
```
Comment (roastduck) [inline python/freetensor/core/RPCTools/center.py:1]: What is `email.header`? It seems `make_header` is not used.

File: python/freetensor/core/RPCTools/core.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+
+# def remove_host(uid):
+#     print("Host %s removed" % uid)
+# def add_host(uid, status):
+#     print("Added a host %s with status %s" % (uid, str(status)))
+# def task_submit(r_uid, l_uid, task):
+#     print("Submit a task from %s to %s" % (r_uid, l_uid))
+# def result_submit(r_uid, l_uid, result):
+#     print("Submit result from %s to %s" % (r_uid, l_uid))
```
Comment (roastduck) [inline python/freetensor/core/RPCTools/core.py:104]: Please remove the commented code. If it was a test, move it to `test/`.

File: python/freetensor/core/RPCTools/core.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            break
+        if server:
+            return server
+        else:
+            raise Exception("Error failed to connect")
+
+    def get_address(self):
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.bind(('', 0))
+        s.connect(('8.8.8.8', 80))
```
Comment (roastduck) [inline python/freetensor/core/RPCTools/core.py:58]: What's that?

Comment (hjsjhn) [reply python/freetensor/core/RPCTools/core.py:64]: I use it to get the server's IP address and an available port.

**Pull request closed without merging**